### PR TITLE
Extract common JS logic from HotReload files

### LINKED ISSUE #2539: [Enhancement] TAE: Extract `hot_reload.js` logic from `hot_reload.rb` and refactor JavaScript files
The goal of this task is extracting the JavaScript logic from `hot_reload.rb` and promote reusability across our JavaScript files.

We might split the JavaScript logic into multiple JavaScript files to compose them. We may JavaScript files in a new directory (`lib/shopify_cli/theme/dev_server/hot_reload/resources*`), so we may end up with something like this:
- `lib/shopify_cli/theme/dev_server/hot_reload/resources/sse_client.js` (we may have more files like this one, which will be shared between `theme.js` and `theme_extension.js`)
- `lib/shopify_cli/theme/dev_server/hot_reload/resources/theme.js`
- `lib/shopify_cli/theme/dev_server/hot_reload/resources/theme_extension.js`

### PR DESCRIPTION
**Pull request by ulgut**
Need to merge de-couple PR first 😅 
I created/modified 4 main files:
- `sse_client.js`: Stores all logic related to client connections and initial message parsing (I assume we can create `ws_client.js` in future).
- `hot_reload.js`: Stores all shared hot reload logic.
- `theme.js`: All `theme serve`-related JS code
- `theme_extension.js`: All `extension serve`-related JS code

### CODE REVIEW

File: lib/shopify_cli/theme/dev_server/hot_reload/resources/hot_reload.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    static reloadCssFile = (filename) => {
+        // Find a stylesheet link starting with /assets (locally-served only) containing the filename
+        let links = document.querySelectorAll(`link[href^="/assets"][href*="${filename}"][rel="stylesheet"]`);
+
+        Array.from(links).forEach((link) => {
+            if (!link) {
+                console.log(`[HotReload] Could not find link for stylesheet ${filename}`);
+            } else {
+                console.log(link)
```
Comment (jessetuglu) [inline lib/shopify_cli/theme/dev_server/hot_reload/resources/hot_reload.js:39]: let me know what your thoughts are on logging these urls – I think we could maybe avoid logging here

**Review by karreiro — COMMENTED** [review]
Thank you, @jessetuglu! Awesome stuff! 🚀 

I've noticed some minor things, so I created this PR (https://github.com/Shopify/shopify-cli/pull/2567) to centralize them. To summarize, it:
- It formats with [Prettier](4187a97a081d15a673da9c7a9766bd46a9893837) the JavaScript files in the `lib/shopify_cli/theme/dev_server/hot_reload/resources/*` directory (I will setup a linter when we move the codebase to the monorepo)
- It removes `SSEClient` and `HotReload` from the [global scope](https://github.com/Shopify/shopify-cli/pull/2567/files#diff-a00fdc42fd6180aeb4529ecda9e64237e66af79af446352bdcebfc4647634a8cR21)
- It shares a bit more logic between the [`ReloadJSHook`](https://github.com/Shopify/shopify-cli/pull/2567/files#diff-f1ccc8ead6ce96e6345f70ffc7450465ae5d52b0f899b8be6bc412011ca7b72cR8) classes
- It renames the [`ReloadJSHook` to `ScriptInjector` ](https://github.com/Shopify/shopify-cli/pull/2553#pullrequestreview-1080246173)

Let me know wdyt. If you agree with those changes I believe we may merge https://github.com/Shopify/shopify-cli/pull/2567 into this branch :)

**Review by karreiro — APPROVED** [review]
Thank you, @jessetuglu! 🚀

**Review by mgmanzella — APPROVED** [review]
niiice 🔥

**Pull request merged**